typespec-extension/src/code-model-builder.ts

```diff
@@ -1696,275 +1692,200 @@ export class CodeModelBuilder {
-        // integer
-        const integerSize = scalarName === "safeint" || scalarName.includes("int64") ? 64 : 32;
-        return this.processIntegerSchema(type, nameHint, integerSize);
-      } else if (scalarName.startsWith("float")) {
-        // float point
-        return this.processNumberSchema(type, nameHint);
-      } else {
-        throw new Error(`Unrecognized scalar type: '${scalarName}'.`);
-      }
-    } else {
-      const knownValues = getKnownValues(this.program, type as Scalar);
-      if (knownValues) {
-        // use it for extensible enum
-        return this.processChoiceSchema(knownValues, this.getName(type), false);
-      } else {
-        const encode = getEncode(this.program, type);
-        if (encode) {
-          // process as encode
-          if (encode.encoding === "seconds" && hasScalarAsBase(type, "duration")) {
-            return this.processDurationSchema(type, nameHint, getDurationFormat(encode));
-          } else if (
-            (encode.encoding === "rfc3339" || encode.encoding === "rfc7231" || encode.encoding === "unixTimestamp") &&
-            (hasScalarAsBase(type, "utcDateTime") || hasScalarAsBase(type, "offsetDateTime"))
-          ) {
-            if (encode.encoding === "unixTimestamp") {
-              return this.processUnixTimeSchema(type, nameHint);
-            } else {
-              return this.processDateTimeSchema(type, nameHint, encode.encoding === "rfc7231");
-            }
-          } else if (encode.encoding === "base64url" && hasScalarAsBase(type, "bytes")) {
-            return this.processByteArraySchema(type, nameHint, true);
-          }
-        }
-
-        if (type.baseScalar) {
-          // fallback to baseScalar
-          const schema = this.processScalar(type.baseScalar, getFormat(this.program, type), nameHint);
-          const doc = getDoc(this.program, type);
-          const summary = getSummary(this.program, type);
-          if (doc) {
-            schema.language.default.description = doc;
-          }
-          if (summary) {
-            schema.summary = summary;
-          }
-          return schema;
-        } else {
-          throw new Error(`Unrecognized scalar type: '${scalarName}'.`);
-        }
+        case "uri":
+          return this.processUrlSchemaFromSdkType(type, nameHint);
       }
     }
   }
 
-  private processAnySchema(type: IntrinsicType, name: string): AnySchema {
+  private processAnySchemaFromSdkType(): AnySchema {
     return this.anySchema;
   }
 
-  private processStringSchema(type: Scalar, name: string): StringSchema {
+  private processStringSchemaFromSdkType(type: SdkBuiltInType, name: string): StringSchema {
     return this.codeModel.schemas.add(
-      new StringSchema(name, this.getDoc(type), {
-        summary: this.getSummary(type),
+      new StringSchema(name, this.getDoc(type.__raw as Type), {
+        summary: this.getSummary(type.__raw as Type),
```

Comment:
Why do you need these `as`?

Reply:
type.__raw is `Type | undefined`. Maybe better change the getDoc() and getSummary()  method's signature to accept `Type | undefined`.

Reply:
On the case that you have to do e.g. `__raw as Union`, you may want to do an assert first e.g. `assert __raw.kind === "Union"; Union type = __raw as Union;`.